code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -882,29 +882,43 @@ obj/item/assembly/radio_horn/receive_signal()
 	var/thingsneeded = null
 	var/obj/item/ammo/bullets/result = null
 	var/obj/item/accepteditem = null
+	var/craftname = null
+	var/obj/item/material = null
```

Comment:
No need to define this here, just put it inside the `proc/craftwith(...)`. Unless you plan on making pipeshot assemblies recyclable in the future.